src/meds_torch/predict.py

```diff
@@ -0,0 +1,285 @@
+    └────────┴──────────────┴──────────────────────────────┘
+    >>> del predictions[1]['1d']
+    >>> import pytest
+    >>> with pytest.raises(RuntimeError):
+    ...     pytest.raises(process_predictions(predictions, keys))
+    """
+    predict_df = pl.DataFrame()
+
+    for key in model_keys:
+        if key not in predictions[0]:
+            continue
+
+        if not isinstance(predictions[0][key], torch.Tensor):
+            continue
+
+        if len(predictions[0][key].shape) == 0:  # skip scalars
+            continue
+
+        key_name = model_keys[key]
+        try:
+            # Process the tensor into appropriate format
+            processed_data = process_tensor_batches(predictions, key)
+
+            # Create a Polars series from the processed data
+            predict_df = predict_df.with_columns(pl.Series(processed_data).alias(key_name))
+        except Exception as e:
+            raise RuntimeError(f"Error processing key {key}: {str(e)}")
+
+    return predict_df
+
+
+@task_wrapper
+def predict(cfg: DictConfig, datamodule=None) -> tuple[dict[str, Any], dict[str, Any]]:
+    """Evaluates given checkpoint on a datamodule testset.
+
+    This method is wrapped in optional @task_wrapper decorator, that controls the
+    behavior during failure. Useful for multiruns, saving info about the crash, etc.
+
+    Args:
+        cfg: DictConfig configuration composed by Hydra.
+    Returns:
+        Tuple[dict, dict] with metrics and dict with all instantiated objects.
+    """
+    assert cfg.ckpt_path
+    if not cfg.data.do_include_subject_id:
+        raise ValueError("Subject ID is required for generating trajectories")
+    if not cfg.data.do_include_prediction_time:
+        raise ValueError("Prediction time is required for generating trajectories")
+
+    log.info(f"Instantiating datamodule <{cfg.data._target_}>")
+    if not datamodule:
+        datamodule: LightningDataModule = hydra.utils.instantiate(cfg.data)
+
+    log.info(f"Instantiating model <{cfg.model._target_}>")
+    model: LightningModule = hydra.utils.instantiate(cfg.model)
+    model.load_state_dict(torch.load(cfg.ckpt_path)["state_dict"])
+
+    log.info("Instantiating loggers...")
+    logger: list[Logger] = instantiate_loggers(cfg.get("logger"))
+
+    log.info(f"Instantiating trainer <{cfg.trainer._target_}>")
+    trainer: Trainer = hydra.utils.instantiate(cfg.trainer, logger=logger)
+
+    object_dict = {
+        "cfg": cfg,
+        "datamodule": datamodule,
+        "model": model,
+        "logger": logger,
+        "trainer": trainer,
+    }
+
+    if logger:
+        log.info("Logging hyperparameters!")
+        log_hyperparameters(object_dict)
+
+    log.info("Starting Generating Predictions!")
+    predictions = trainer.predict(model=model, dataloaders=datamodule)
+
+    # Extract input trajectory
+    dfs = {}
+
+    subject_ids = pl.Series(list(chain.from_iterable([batch["subject_id"] for batch in predictions]))).cast(
```

Comment:
I suspect there is a cleaner way of doing this where you construct one larger numpy array first then build a polars dataframe from that in one shot, but I don't think it particularly matters.